### PR TITLE
finish the logic part of the calculator

### PR DESCRIPTION
**Pull request by majdalkilany**
Add the logic for the calculator 
all the operations working fine

### CODE REVIEW

File: src/components/calculator/calculator.js [hunk 1]
```diff
@@ -25,28 +26,31 @@ const buttons = [
   '=',
 ];
 
+const initialObject = { total: 0, next: null, operation: null };
 const Calculator = () => {
-  const [calculatorInputs, setCalculatorInputs] = useState('0');
+  // const [calculatorInputs, setCalculatorInputs] = useState('0');
```
Comment (V-Blaze) [inline src/components/calculator/calculator.js]: - - You have done great so far. :+1:  Your calculator app works great but there are a few tasks you need to complete before moving to the next milestone. :+1: 
- [ ] It is against the best practices to leave unused code as a comment in your codebase, I'll suggest you remove this as it makes your codebase a bit unprofessional. :smile: :wave:

File: src/components/calculator/calculator.js [hunk 2]
```diff
@@ -25,28 +26,31 @@ const buttons = [
...
-    }
-    if (value === 'AC') {
-      setCalculatorInputs('0');
-    } else {
-      setCalculatorInputs(newCalculatorInputs + value);
-    }
+
+    const result = calculate(obj, value);
+    setObj(result);
+    console.log(result);
```
Comment (V-Blaze) [inline src/components/calculator/calculator.js]: - [ ] You can use `console log`  when debugging but following the best practices, you are meant to remove this once you are done and not push it to your GitHub, I'll suggest you do so. :+1:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @majdalkilany , 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Used provided calculator logic correctly✔
- Calculator works👋 ✔

 ## Required Changes ♻️
- [ ] Your readme file has some issues, it seems you have some syntax error that makes the whole content clipped under client. I'll suggest I'll you rectify this as it makes your readme look unprofessional. :+1: 

- - Screenshot:
 
Screenshot from 2023-07-13 11-56-23


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by majdalkilany** [comment]
fixed all the requested changes

**Review by Reem-lab — APPROVED** [review]
# Hi @majdalkilany,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [x]  No Linters errors 🔥
- [x]  Great job working on the logic💯
- [ ]  Followed GitFlow ⭐🚀
- [ ] Descriptive PR 🚀
- [ ] All operations work as expected 💯💯


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**